jp2/tests/parse_tests.rs

```diff
@@ -1,44 +1,821 @@
 use std::{fs::File, io::BufReader, path::Path};
 
-use jp2::{decode_jp2, ColourSpecificationMethods, JBox as _, JP2File};
+use jp2::{decode_jp2, BitDepth, ChannelTypes, ColourSpecificationMethods, JBox as _, JP2File};
 
 struct ExpectedConfiguration {
+    compatibility_list: Vec<String>,
     width: u32,
     height: u32,
     num_components: u16,
     bit_depth: u8,
-    grayscale: bool,
+    colourspace: u32,
+    colour_specification_method: ColourSpecificationMethods,
```

Comment:
I'm not sure I see the point of the ExpectedConfiguration, as it will expand to cover all the properties of the JP2 container so it only creates an abstraction over the API which is easily refactored with most IDEs. For example we could add XML/UUID boxes, but we don't, why? We could probably use JPXML to verify a lot of the boxes, this may make it more approachable for testing (it's only XML) and dogfoods the JPXML part too. Thoughts?

Reply:
For the JP2 tests, most of the sample data  has the same values. I don't really want to copy the same test conditions, and this was a way to handle things that are a bit more deeply nested.

JPXML is an option, but that would be an dependency outside the module, and its not always the easiest thing to do comparisons on (e.g. whitespace differences).

Reply:
That's a fair point on the whitespace differences, although with xpath it should be less of an issue. You're right on the external dependency, perhaps we can use it for the integration tests. Lets see how far we can take the current approach but given a large enough test samples we'll need to cater for all scenarios. Ideally we would have the minimum amount of files that provides the maximum amount of coverage, but that is probably difficult to create.